allennlp/modules/seq2vec_encoders/pytorch_seq2vec_wrapper.py

```diff
@@ -68,48 +67,25 @@ def forward(self,  # pylint: disable=arguments-differ
             # at the end of the max sequence length, so we have to use the state of the RNN below.
             return self._module(inputs, hidden_state)[0][:, -1, :]
 
-        # In some circumstances you may have sequences of zero length. For example, if this is the
-        # embedding layer for a character-based encoding, the original input will have size
-        #   (batch_size, max_sentence_length, max_word_length, encoding_dim)
-        # and then ``TimeDistributed`` will reshape it to
-        #   (batch_size * max_sentence_length, max_word_length, encoding_dim)
-        # in which case all the rows corresponding to word padding will be
-        # "empty character sequences".
-        #
-        # ``pack_padded_sequence`` requires all sequence lengths to be > 0, so here we
-        # adjust the ``mask`` so that every sequence has length at least 1. Then after
-        # running the RNN we zero out the corresponding rows in the result.
-
-        # First count how many sequences are empty.
-        batch_size = mask.size()[0]
-        num_valid = torch.sum(mask[:, 0]).int().data[0]
-
-        # Force every sequence to be length at least one. Need to `.clone()` the mask
-        # to avoid a RuntimeError from shared storage.
-        if num_valid < batch_size:
-            mask = mask.clone()
-            mask[:, 0] = 1
-
-        sequence_lengths = get_lengths_from_binary_sequence_mask(mask)
-        sorted_inputs, sorted_sequence_lengths, restoration_indices = sort_batch_by_length(inputs,
-                                                                                           sequence_lengths)
-
-        packed_sequence_input = pack_padded_sequence(sorted_inputs,
-                                                     sorted_sequence_lengths.data.tolist(),
-                                                     batch_first=True)
+        batch_size = mask.size(0)
 
-        # Actually call the module on the sorted PackedSequence.
-        _, state = self._module(packed_sequence_input, hidden_state)
+        _, state, restoration_indices, num_valid = \
+            self.sort_and_run_forward(self._module, inputs, mask, hidden_state)
 
         # Deal with the fact the LSTM state is a tuple of (state, memory).
         if isinstance(state, tuple):
             state = state[0]
 
-        # We sorted by length, so if there are invalid rows that need to be zeroed out
-        # they will be at the end. Note that at this point batch_size is the second
-        # dimension of state.
+        # Add back invalid rows.
         if num_valid < batch_size:
-            state[:, num_valid:, :] = 0.
+            # batch size is the second dimension here, because pytorch
+            # returns RNN state as a tensor of shape (num_layers * num_directions,
+            # batch_size, hidden_size)
+            num_layers_times_directions, _, encoding_dim = state.size()
```

Comment:
You can replace `_` with `num_valid` here, so you don't have to return it from `sort_and_run_forward`.